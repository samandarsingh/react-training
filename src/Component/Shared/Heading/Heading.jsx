import React from "react";
import PropTypes from "prop-types";
import styles from "./Heading.module.scss";
/**
 * 
 * @param {string} headingType 
 * @param {string} headingText
 * @return Element
 */
const Heading = props => {
    const {headingType,headingText} = props;
    switch(headingType){
        case "h1":
            return <h1 className={styles.headingItem}>{headingText}</h1>;
        case "h2":
            return <h2 className={styles.headingItem}>{headingText}</h2>;
        case "h3":
            return <h3 className={styles.headingItem}>{headingText}</h3>;
        case "h4":
            return <h4 className={styles.headingItem}>{headingText}</h4>;
        default:
            return null;        
    }   
}
Heading.propTypes = {
    headingText: PropTypes.string.isRequired,
    headingType: PropTypes.string.isRequired
}
Heading.defaultProps = {
    headingType: "h4"
}
export default Heading;