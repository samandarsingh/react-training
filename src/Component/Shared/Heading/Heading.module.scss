.headingItem {    
    h4 {
        color: azure;
        text-decoration: none;        
        font-size: 16px;
        text-align: -webkit-left;
        margin-left: 30px;        
    }
}