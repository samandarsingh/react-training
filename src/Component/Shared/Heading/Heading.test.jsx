import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import Heading from "./Heading";

describe(`my component testing`,() => {
    const props = {
        headingText: "",
        headingType:"h4"
    }
    const wrapper = mount(<Heading {...props}/>);    
    it(`my component should be run`,() => {        
        expect(wrapper).toBeTruthy();
    }); 
    it(`testing heading type should be h4`, () => {        
        expect(
            wrapper
            .find(Heading)
            .props()
            .headingType
        ).toEqual(`h4`);

        expect(
            wrapper
            .find(Heading)
            .find(`h4`)
            .hasClass(`headingItem`)
        ).toEqual(true);
    });
    it(`testing change props headingType should be render`, () => {
        wrapper
        .setProps(
            { 
                headingType:"h1"
            }
        );        
        expect(
            wrapper
            .find(Heading)
            .props()
            .headingType
        ).toEqual(`h1`);
        wrapper
        .setProps(
            { 
                headingType:"h2"
            }
        );
        expect(
            wrapper
            .find(Heading)
            .props()
            .headingType
        ).toEqual(`h2`);
        wrapper
        .setProps(
            { 
                headingType:"h3"
            }
        );        
        expect(
            wrapper
            .find(Heading)
            .props()
            .headingType
        ).toEqual(`h3`);
    })
    it(`testing change props headingType should be render`, () => {        
        wrapper
        .setProps(
            { 
                headingType:""
            }
        );
        expect(
            wrapper
            .find(Heading)
            .props()
            .headingType
        ).toEqual(``);
    });
    it(`renders correctly`, () => {        
        const tree = renderer
          .create(<Heading {...props}/>)
          .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
