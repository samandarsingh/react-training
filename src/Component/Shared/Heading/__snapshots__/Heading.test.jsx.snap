// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my component testing renders correctly 1`] = `
<h4
  className="headingItem"
>
  
</h4>
`;
